app/controllers/api/configuration_script_sources_controller.rb

```diff
@@ -20,12 +20,16 @@ def delete_resource(type, id, _data = {})
 
     def create_resource(_type, _id, data)
       validate_attrs(data)
+
       manager_id = parse_id(data['manager_resource'], :providers)
       raise 'Must specify a valid manager_resource href or id' unless manager_id
       manager = resource_search(manager_id, :providers, collection_class(:providers))
-      type = ConfigurationScriptSource.class_for_manager(manager)
-      raise "ConfigurationScriptSource cannot be added to #{manager_ident(manager)}" unless type.respond_to?(:create_in_provider_queue)
-      task_id = type.create_in_provider_queue(manager.id, data.except('manager_resource').deep_symbolize_keys)
+
+      type = "#{manager.type}::ConfigurationScriptSource"
+      klass = ConfigurationScriptSource.class_from_string(type)
```

Comment:
I had to look at the tests to know what class_from_string did.

`descendent_class_from_name`

Naming is hard.  Technically, it should be descendent or self but that's just too many words.

I like the idea, just not sure about the name.